code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -570,43 +651,13 @@
 	background_icon_state = "bg_alien"
 
 /datum/action/innate/linked_speech/Activate()
-	var/mob/living/carbon/human/H = owner
-	var/datum/species/oozeling/stargazer/species = target
-	if(!species || !(H in species.linked_mobs))
-		to_chat(H, "<span class='warning'>The link seems to have been severed...</span>")
-		Remove(H)
+	var/message = tgui_input_text(usr, "Message:", "Slime Telepathy")
+	if(!message || !length(message))
 		return
-
-	var/message = stripped_input(usr, "Message:", "Slime Telepathy")
 	if(CHAT_FILTER_CHECK(message))
 		to_chat(usr, "<span class='warning'>Your message contains forbidden words.</span>")
 		return
-	message = H.treat_message_min(message)
-	if(!species || !(H in species.linked_mobs))
-		to_chat(H, "<span class='warning'>The link seems to have been severed...</span>")
-		Remove(H)
-		return
-
-	if(QDELETED(H) || H.stat == DEAD)
-		species.unlink_mob(H)
-		return
-
-	var/mob/living/carbon/human/star_owner = species.slimelink_owner.resolve()
-
-	if(message && star_owner)
-		var/msg = "<i><font color=#008CA2>\[[star_owner.real_name]'s Slime Link\] <b>[H]:</b> [message]</font></i>"
-		log_directed_talk(H, star_owner, msg, LOG_SAY, "slime link")
-		for(var/X in species.linked_mobs)
-			var/mob/living/M = X
-			if(QDELETED(M) || M.stat == DEAD)
-				species.unlink_mob(M)
-				continue
-			to_chat(M, msg)
-
-		for(var/X in GLOB.dead_mob_list)
-			var/mob/M = X
-			var/link = FOLLOW_LINK(M, H)
-			to_chat(M, "[link] [msg]")
+	usr.say(".[MODE_KEY_SLIMELINK] [message]")
```

Comment:
does say already perform ic filtering and message treating?

Reply:
Yes. https://github.com/BeeStation/BeeStation-Hornet/blob/master/code/modules/mob/living/say.dm#L81-L94